pandas/core/algorithms.py

```diff
@@ -911,10 +911,15 @@ def select_n_frame(frame, columns, n, method, keep):
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
+    tmp = Series(frame.index)
+    frame.reset_index(inplace=True, drop=True)
     ser = getattr(frame[columns[0]], method)(n, keep=keep)
     if isinstance(ser, Series):
         ser = ser.to_frame()
```

Comment:
right I don't think it IS there. what you can easily do it simply use the routine to select the nsmallest/largest (its according to a single column), *then* sort accordting to all of the columns provided (this is still quite efficient as the selection uses partition sort which).